ui/packages/catalog-ui-search/src/main/webapp/react-component/utils/validation/validation.tsx

```diff
@@ -270,68 +271,134 @@ function validateLinePolygon(mode: string, currentValue: string) {
+    hemisphere: lowerRight.hemisphere,
+    northPole: lowerRight.hemisphere.toUpperCase() === 'NORTHERN',
+  }
+  upperLeftParts.northing =
+    upperLeftParts.zoneNumber === 0 || upperLeftParts.northPole
+      ? upperLeftParts.northing
+      : upperLeftParts.northing - NORTHING_OFFSET
+  lowerRightParts.northing =
+    lowerRightParts.zoneNumber === 0 || lowerRightParts.northPole
+      ? lowerRightParts.northing
+      : lowerRightParts.northing - NORTHING_OFFSET
+  const north = Number(converter.UTMUPStoLL(upperLeftParts).lat.toFixed(5))
+  const south = Number(converter.UTMUPStoLL(lowerRightParts).lat.toFixed(5))
+  const west = Number(converter.UTMUPStoLL(upperLeftParts).lon.toFixed(5))
+  const east = Number(converter.UTMUPStoLL(lowerRightParts).lon.toFixed(5))
+  return { north, south, west, east }
+}
+
+function validateLatitudes(north: any, south: any, isUsngOrUtmUps: boolean) {
+  if (!isNaN(south) && !isNaN(north) && south >= north) {
+    return {
+      error: true,
+      message: isUsngOrUtmUps
+        ? 'Upper left coordinate must be located above lower right coordinate'
+        : 'North value must be greater than south value',
     }
-    lowerRightParts.northing =
-      lowerRightParts.zoneNumber === 0 || northPole
-        ? lowerRightParts.northing
-        : lowerRightParts.northing - NORTHING_OFFSET
-    south = Number(converter.UTMUPStoLL(lowerRightParts).lat.toFixed(5))
-  } else {
-    north = Number(value.north)
-    south = Number(value.south)
   }
-  if (south !== null && north !== null && south >= north) {
+  return initialErrorState
+}
+
+function validateLongitudes(west: any, east: any, isUsngOrUtmUps: boolean) {
+  if (!isNaN(west) && !isNaN(east) && west === east) {
     return {
       error: true,
-      message:
-        value.isUsng || value.isUtmUps
-          ? 'Upper left coordinate must be located above lower right coordinate'
-          : 'North value must be greater than south value',
+      message: isUsngOrUtmUps
+        ? 'Left bound cannot equal right bound'
+        : 'West value cannot equal east value',
+    }
+  }
+  return initialErrorState
+}
+
+function validateBoundingBox(key: string, value: any) {
+  const { north, south, west, east } = value.isDms
+    ? getDmsCoords(value)
+    : value.isUsng
+      ? getUsngCoords(value.upperLeft, value.lowerRight)
+      : value.isUtmUps
+        ? getUtmUpsCoords(value.upperLeft, value.lowerRight)
+        : getDdCoords(value)
+  const isUsngOrUtmUps = value.isUsng || value.isUtmUps
+  if (key.toLowerCase().includes('lon')) {
+    const { error, message } = validateLongitudes(west, east, isUsngOrUtmUps)
+    if (error) {
+      return { error, message }
+    }
+    return validateLatitudes(north, south, isUsngOrUtmUps)
+  } else {
+    const { error, message } = validateLatitudes(north, south, isUsngOrUtmUps)
+    if (error) {
+      return { error, message }
     }
+    return validateLongitudes(west, east, isUsngOrUtmUps)
```

Comment:
❓ why is `validateLongitudes` being called here?

Reply:
If the key is 'lat' but there are no issues with the latitudes, we still want to display an error message if east = west, which is what validateLongitudes checks for